pkg/resources/stream_on_external_table_acceptance_test.go

```diff
@@ -0,0 +1,586 @@
+					assert.Check(resource.TestCheckResourceAttr(resourceName, "describe_output.0.mode", string(sdk.StreamModeInsertOnly))),
+					assert.Check(resource.TestCheckResourceAttrSet(resourceName, "describe_output.0.stale_after")),
+					assert.Check(resource.TestCheckResourceAttr(resourceName, "describe_output.0.owner_role_type", "ROLE")),
+				),
+			},
+			// import
+			{
+				Config:       config.FromModel(t, modelWithExtraFieldsModified),
+				ResourceName: resourceName,
+				ImportState:  true,
+				ImportStateCheck: assert.AssertThatImport(t,
+					resourceassert.ImportedStreamOnExternalTableResource(t, resourceId).
+						HasNameString(id.Name()).
+						HasDatabaseString(id.DatabaseName()).
+						HasSchemaString(id.SchemaName()).
+						HasFullyQualifiedNameString(id.FullyQualifiedName()).
+						HasInsertOnlyString(r.BooleanTrue).
+						HasExternalTableString(externalTable.ID().FullyQualifiedName()).
+						HasCommentString("bar"),
+				),
+			},
+		},
+	})
+}
+
+func TestAcc_StreamOnExternalTable_CopyGrants(t *testing.T) {
+	_ = testenvs.GetOrSkipTest(t, testenvs.EnableAcceptance)
+	acc.TestAccPreCheck(t)
+	id := acc.TestClient().Ids.RandomSchemaObjectIdentifier()
+	resourceName := "snowflake_stream_on_external_table.test"
+
+	var createdOn string
+
+	stageID := acc.TestClient().Ids.RandomSchemaObjectIdentifier()
+	stageLocation := fmt.Sprintf("@%s", stageID.FullyQualifiedName())
+	_, stageCleanup := acc.TestClient().Stage.CreateStageWithURL(t, stageID)
+	t.Cleanup(stageCleanup)
+
+	externalTable, externalTableCleanup := acc.TestClient().ExternalTable.CreateWithLocation(t, stageLocation)
+	t.Cleanup(externalTableCleanup)
+
+	model := model.StreamOnExternalTable("test", id.DatabaseName(), externalTable.ID().FullyQualifiedName(), id.Name(), id.SchemaName()).WithInsertOnly(r.BooleanTrue)
+	resource.Test(t, resource.TestCase{
+		ProtoV6ProviderFactories: acc.TestAccProtoV6ProviderFactories,
+		TerraformVersionChecks: []tfversion.TerraformVersionCheck{
+			tfversion.RequireAbove(tfversion.Version1_5_0),
+		},
+		CheckDestroy: acc.CheckDestroy(t, resources.StreamOnTable),
+		Steps: []resource.TestStep{
+			{
+				Config: config.FromModel(t, model.WithCopyGrants(true)),
+				Check: assert.AssertThat(t, resourceassert.StreamOnTableResource(t, resourceName).
+					HasNameString(id.Name()),
+					assert.Check(resource.TestCheckResourceAttrWith(resourceName, "show_output.0.created_on", func(value string) error {
+						createdOn = value
+						return nil
+					})),
+				),
+			},
+			{
+				Config: config.FromModel(t, model.WithCopyGrants(false)),
+				Check: assert.AssertThat(t, resourceassert.StreamOnTableResource(t, resourceName).
+					HasNameString(id.Name()),
+					assert.Check(resource.TestCheckResourceAttrWith(resourceName, "show_output.0.created_on", func(value string) error {
+						if value != createdOn {
+							return fmt.Errorf("view was recreated")
+						}
+						return nil
+					})),
+				),
+			},
+			{
```

Comment:
Shouldn't this test have at least one test case where attribute casing CREATE OR REPLACE to be called?

Reply:
Added.